tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -159,6 +159,51 @@ def test_write(self, cloud_storage_type):
             for file, count in index_segment_pair.items():
                 assert count == 2, f'expected one index and one log for {file}, found {count}'
 
+    @skip_debug_mode
+    @cluster(num_nodes=4)
+    def test_recover(self):
+        # Recovery will leave behind results files, which aren't tracked.
```

Comment:
The result files should be periodically removed by the recovery process. If they are left behind after recovery exits it seems like a bug.

Reply:
You're right, this was a bug with how topic recovery state was being reported, and how RPK was polling. Added a fix to RPK to make it wait for an `incomplete` status, and updated redpanda to make the recovery state toggle to be atomic.